packages/snaps-controllers/src/snaps/endowments/name-lookup.ts

```diff
@@ -128,20 +156,53 @@ export function getChainIdsCaveat(
     return null;
   }
 
-  assert(permission.caveats.length === 1);
-  assert(permission.caveats[0].type === SnapCaveatType.ChainIds);
+  const caveat = permission.caveats.find(
+    (permCaveat) => permCaveat.type === SnapCaveatType.ChainIds,
+  );
+  assert(permission.caveats.length === 1 || permission.caveats.length === 2);
+  assert(caveat);
+  assert(caveat.type === SnapCaveatType.ChainIds);
```

Comment:
I don't think we need to assert the type when we use `find()`

Reply:
you're right.